src/utils/RequestQueue.ts

```diff
@@ -119,23 +140,31 @@ export default class RequestQueue {
    *  until the request is resolved or cancelled.
    *  Note that the return type of the promise will match that of the first request's instance.
    */
-  public addRequest<T>(key: string, requestAction: () => Promise<T>, delayMs = 0): Promise<T> {
+  public addRequest<T>(key: string, requestAction: () => Promise<T>, lowPriority = false, delayMs = 0): Promise<T> {
     if (!this.allRequests.has(key)) {
       // New request!
       const requestItem = this.registerRequest(key, requestAction);
       // If a delay is set, wait to add this to the queue.
       if (delayMs > 0) {
-        const timeoutId = setTimeout(() => this.addRequestToQueue(key), delayMs);
+        const timeoutId = setTimeout(() => this.addRequestToQueue(key, lowPriority), delayMs);
         // Save timeout information to request metadata
         requestItem.timeoutId = timeoutId;
       } else {
         // No delay, add immediately
+        this.addRequestToQueue(key, lowPriority);
+      }
+    } else {
+      const lowPriorityIndex = this.queueLowPriority.indexOf(key);
+      if (lowPriorityIndex > -1 && !lowPriority) {
+        // This request is registered and queued, but is now being requested with high priority.
+        // Promote it to high priority.
+        this.queueLowPriority.splice(lowPriorityIndex, 1);
         this.addRequestToQueue(key);
+      } else if (delayMs <= 0) {
```

Comment:
Might be worth thinking more deeply about how "promote to high priority" and "cancel delay and execute immediately" might interact here.